### PR TITLE
Feature/contacts

### PR DESCRIPTION
**Pull request by SunnyCat2013**

### CODE REVIEW

File: react-redux/contactsMVC/components/Conversation.jsx
Comment (ZhengHe-MD) [inline react-redux/contactsMVC/components/Conversation.jsx]:
``` javascript
import { Card, CardText } from 'material-ui/Card';
import { List } from 'material-ui/List';
import React, { PropTypes } from 'react';
import ReactDOM from 'react-dom';
import Subheader from 'material-ui/Subheader';
import TextField from 'material-ui/TextField';
import FlatButton from 'material-ui/FlatButton';
```

带括号的先(specific import)，不带的后(default import)。仅仅为了好看一些

File: react-redux/contactsMVC/components/Conversation.jsx
Comment (ZhengHe-MD) [inline react-redux/contactsMVC/components/Conversation.jsx]: don't use global id, use `node-uuid`

Comment (ZhengHe-MD) [reply react-redux/contactsMVC/components/Conversation.jsx]: generage uuid for each comment

``` sh
npm install node-uuid --save
```

``` javascript
import uuid from 'node-uuid';
const nextCommentId = uuid.v4();
```

File: react-redux/contactsMVC/components/Conversation.jsx
Comment (ZhengHe-MD) [inline react-redux/contactsMVC/components/Conversation.jsx]: you can also

``` javascript
import React, { Component, PropTypes } from 'react';
class Conversation extends Component {}
```

File: react-redux/contactsMVC/components/Conversation.jsx
Comment (ZhengHe-MD) [inline react-redux/contactsMVC/components/Conversation.jsx]: name all your event handler as `onDoSomething`.
in this specific case, use `onChangeInputValue`

Comment (ZhengHe-MD) [reply react-redux/contactsMVC/components/Conversation.jsx]: or you can use `onSomethingDone`
in this specific case, use `onInputValueChanged`
pick whichever you like

File: react-redux/contactsMVC/components/Conversation.jsx
Comment (ZhengHe-MD) [inline react-redux/contactsMVC/components/Conversation.jsx]: do this when there are more than two props

``` javascript
<CommentBox
  key={commentObject.id}
  commentObject={commentObject}
  store={this.props.store}
/>
```

File: react-redux/contactsMVC/components/Conversation.jsx
Comment (ZhengHe-MD) [inline react-redux/contactsMVC/components/Conversation.jsx]: rename all your **show_comments** to **showComments**

File: react-redux/contactsMVC/components/Conversation.jsx
Comment (ZhengHe-MD) [inline react-redux/contactsMVC/components/Conversation.jsx]:
``` javascript
{
  showComments.length > 0 ?
    this.getShowComments(showComments) :
    <div><h1>There is not comment.</h1></div>
}
```

or 

``` javascript
{
  showComments.length > 0
    ? this.getShowComments(showComments)
    : <div><h1>There is not comment.</h1></div>
}
```

File: react-redux/contactsMVC/components/Conversation.jsx
Comment (ZhengHe-MD) [inline react-redux/contactsMVC/components/Conversation.jsx]: event handler naming convention

File: react-redux/contactsMVC/components/Conversation.jsx
Comment (ZhengHe-MD) [inline react-redux/contactsMVC/components/Conversation.jsx]: put **.catch** to a new line

File: react-redux/contactsMVC/components/Conversation.jsx
Comment (ZhengHe-MD) [inline react-redux/contactsMVC/components/Conversation.jsx]: remove all your console.log

File: react-redux/contactsMVC/components/commentbox.jsx
Comment (ZhengHe-MD) [inline react-redux/contactsMVC/components/commentbox.jsx]: remove console

File: react-redux/contactsMVC/components/commentbox.jsx
Comment (ZhengHe-MD) [inline react-redux/contactsMVC/components/commentbox.jsx]: you can do this when all variable names are short. But when they are longer, you may

``` javascript
const {
  id,
  comment,
  username,
  thumbnail,
} = this.props.commentObject
```

File: react-redux/contactsMVC/components/commentbox.jsx
Comment (ZhengHe-MD) [inline react-redux/contactsMVC/components/commentbox.jsx]: do you know why you don't have to bind here?

File: react-redux/contactsMVC/components/commentbox.jsx
Comment (ZhengHe-MD) [inline react-redux/contactsMVC/components/commentbox.jsx]: if it is required, put isRequired here

File: react-redux/contactsMVC/contacts.js
Comment (ZhengHe-MD) [inline react-redux/contactsMVC/contacts.js]: no extra spaces in function call

``` javascript
state.filter(s => s.id !== action.id)
```

File: react-redux/contactsMVC/contacts.js
Comment (ZhengHe-MD) [inline react-redux/contactsMVC/contacts.js]: remove your test data

File: react-redux/contactsMVC/contacts.js
Comment (ZhengHe-MD) [inline react-redux/contactsMVC/contacts.js]: put your reducer into another file. you may call it reducer.js in small project.
in large project, you may organise your project in the way introduced here
https://github.com/erikras/ducks-modular-redux

File: react-redux/contactsMVC/contacts.js
Comment (ZhengHe-MD) [inline react-redux/contactsMVC/contacts.js]: Put these lines of code into App.js and import from that

File: react-redux/contactsMVC/package.json
Comment (ZhengHe-MD) [inline react-redux/contactsMVC/package.json]: webpack, babel-loader, expect should be in devDependencies

Comment (ZhengHe-MD) [reply react-redux/contactsMVC/package.json]: use 

``` sh
npm install something --save-dev
```

or 

``` sh
npm install something -D
```

File: react-redux/contactsMVC/webpack.config.js
Comment (ZhengHe-MD) [inline react-redux/contactsMVC/webpack.config.js:15]: remove commented code

**Review by ZhengHe-MD — CHANGES_REQUESTED** [review]
You may merge this PR after you fix my comments.
You still need to incorporate
**react-redux**
**redux-thunk**
to complete this project

File: react-redux/contactsMVC/components/Conversation.jsx
Comment (maicaiyao) [inline react-redux/contactsMVC/components/Conversation.jsx]: better name? showComments is a verb

File: react-redux/contactsMVC/components/Conversation.jsx
Comment (maicaiyao) [inline react-redux/contactsMVC/components/Conversation.jsx]: react component should be pure. async operations should be in action creators.

**Review by maicaiyao — CHANGES_REQUESTED** [review]

File: react-redux/contactsMVC/components/Conversation.jsx
Comment (ZhengHe-MD) [reply react-redux/contactsMVC/components/Conversation.jsx]: nice catch.
**getCurrentComments**
**getCommentsToShow**
anyone is OK

File: react-redux/contactsMVC/components/Conversation.jsx
Comment (ZhengHe-MD) [reply react-redux/contactsMVC/components/Conversation.jsx]: he will use redux-thunk to do this in next PR

File: react-redux/contactsMVC/components/Conversation.jsx
Comment (ZhengHe-MD) [reply react-redux/contactsMVC/components/Conversation.jsx]: You should do this before you merge

**Review by ZhengHe-MD — APPROVED** [review]

**Pull request merged**
